torchtune/data/_messages.py

```diff
@@ -384,12 +397,41 @@ def __call__(self, sample: Mapping[str, Any]) -> Mapping[str, Any]:
                     role="system", content=self.new_system_prompt, masked=True, eot=True
                 )
             )
+
+        is_multimodal = "image" in sample or (
+            "image" in self._column_map and self._column_map["image"] in sample
+        )
+
+        # Add in image stuffs / load from file
+        image_loaded = False
         for message in sample[self._column_map["conversations"]]:
             role = role_map[message["from"]]
+            content = message["value"]
             if role == "system" and self.new_system_prompt is not None:
                 continue
-            content = message["value"]
-            masked = (role != "assistant") and (not self.train_on_input)
+            if role == "user":
+                if is_multimodal and not image_loaded:
```

Comment:
I dont understand this. After we load one image, then if there is some other message with role=="user", we will just ignore it? Shouldnt we raise an error instead of silently ignoring it?

Reply:
this will only load and append image tokens for the first user message. all other messages, including other user messages, will be added as normal.

Reply:
Do you mind making this comment in the code?